src/app/components/tree/tree.ts

```diff
@@ -67,6 +64,7 @@ import { Scroller, ScrollerModule, ScrollerOptions } from 'primeng/scroller';
                     [attr.aria-expanded]="this.node.expanded"
                     [attr.aria-selected]="isSelected()"
                     [attr.aria-label]="node.label"
+                    [attr.data-id]="node.data"
```

Comment:
Unfortunately this is even according to your documentation a "no-go". `TreeNode.data` is of generic type `<T = any> `. You are here assuming its string.

In my projects my `T` is usually and object, and even then it is not Readonly object, but allows mutation.
This simply would not work for anything other that when data is unique string (ie. ID).

This is just not good approach for triggering focus on this element.
Much better solution would be to use `ElementRef` injector (host element) instance and target this `.p-treenode-content` div element via its `this.hostEl.nativeElement.querySelector('.p-treenode-content').focus()`

Reply:
refactored in https://github.com/primefaces/primeng/pull/12485